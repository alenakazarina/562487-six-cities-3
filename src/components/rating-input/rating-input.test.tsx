import * as React from 'react';
import * as renderer from 'react-test-renderer';
import RatingInput from './rating-input';

const mockFn = jest.fn();

describe(`RatingInput`, () => {
  it(`should render RatingInput with rating equals perfect checked`, () => {
    const tree = renderer.create(
        <RatingInput
          rating={{value: 5, title: `perfect`}}
          isChecked={true}
          onChange={mockFn}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`should render RatingInput with rating equals terribly not checked`, () => {
    const tree = renderer.create(
        <RatingInput
          rating={{value: 1, title: `terribly`}}
          isChecked={false}
          onChange={mockFn}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
