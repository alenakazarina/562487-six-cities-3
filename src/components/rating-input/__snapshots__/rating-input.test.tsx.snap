// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingInput should render RatingInput with rating equals perfect checked 1`] = `
Array [
  <input
    checked={true}
    className="form__rating-input visually-hidden"
    id="5-stars"
    name="rating"
    onChange={[MockFunction]}
    type="radio"
    value={5}
  />,
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="5-stars"
    title="perfect"
  >
    <svg
      className="form__star-image"
      height="33"
      width="37"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>,
]
`;

exports[`RatingInput should render RatingInput with rating equals terribly not checked 1`] = `
Array [
  <input
    checked={false}
    className="form__rating-input visually-hidden"
    id="1-stars"
    name="rating"
    onChange={[MockFunction]}
    type="radio"
    value={1}
  />,
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="1-stars"
    title="terribly"
  >
    <svg
      className="form__star-image"
      height="33"
      width="37"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>,
]
`;
